crates/outbound-http/src/lib.rs

```diff
@@ -63,7 +53,15 @@ impl outbound_http::Host for OutboundHttp {
             }
 
             let method = method_from(req.method);
-            let url = Url::parse(&req.uri).map_err(|_| HttpError::InvalidUrl)?;
+            let mut url = Url::parse(&req.uri).map_err(|_| HttpError::InvalidUrl)?;
+
+            if url.host_str() == Some("self.local") {
+                url.set_host(self.origin.host())
```

Comment:
Should we update the scheme here too? Thinking about testing on `http://localhost` and then switching to `https://my.fermyon.app`.

Reply:
yea right now you'd have to change the scheme in the request from the app and that seems like a papercut.

Reply:
Updated the implementation so you don't need to use `self.local` in the uri when making a request and instead you can use relative paths and the correct origin info will be added to the uri as you originally suggested in #1533

Reply:
`self.local` is now `self` because... one word is better than two!

Reply:
We may still want to _allow_ `http://self.local/abc` (in addition to just `/abc`) for code that expects an absolute URL, like (I think?) the Go SDK.

Reply:
I'm not seeing where the Go SDK requires an absolute URL. Could you point me in right direction?

Reply:
I was thinking that since we replace the default http client ([here](https://github.com/fermyon/spin/blob/main/sdk/go/http/http.go#L36)) it might break something in that path if you pass a relative URL, but I haven't tested.